File: 2374.cpp
struct Solution {
    int edgeScore(const vector<int>& edges) {
        vector<long long> scores(edges.size(), 0);
        for (size_t i = 0; i < edges.size(); ++i){
            scores[edges[i]] += i;
        }
        size_t idx = 0;
        for (size_t i = 1; i < edges.size(); ++i){
            if (scores[i] > scores[idx]){
                idx = i;
            }
        }
        return idx;
    }
};

File: sol/2374.py
class Solution:
    def edgeScore(self, edges: List[int]) -> int:
        scores = [0] * len(edges)
        for i, e in enumerate(edges):
            scores[e] += i
        idx = 0
        for i in range(1, len(edges)):
            if scores[i] > scores[idx]:
                idx = i
        return idx
